File: frontend/src/services/authService.ts
import api from "./api";

export interface RegisterRequest {
  fullName: string;
  email: string;
  password: string;
}

export interface LoginRequest {
  email: string;
  password: string;
}

export const register = (data: RegisterRequest) => {
  return api.post("/auth/register", data);
};

export const login = (data: LoginRequest) => {
  return api.post("/auth/login", data);
};
export interface ForgotPasswordRequest {
  email: string;
}

export interface VerifyOtpRequest {
  email: string;
  otp: string;
}

export interface ResetPasswordRequest {
  email: string;
  newPassword: string;
}

export const forgotPassword = (
  data: ForgotPasswordRequest
) => {
  return api.post("/auth/forgot-password", data);
};

export const verifyOtp = (
  data: VerifyOtpRequest
) => {
  return api.post("/auth/verify-otp", data);
};

export const resetPassword = (
  data: ResetPasswordRequest
) => {
  return api.post("/auth/reset-password", data);
};
export const googleLogin = (credential: string) => {
  return api.post("/auth/google", {
    credential,
  });
};

File: backend/src/main/java/com/interviewx/auth/service/AuthService.java
package com.interviewx.auth.service;

import com.interviewx.auth.dto.ForgotPasswordRequest;
import com.interviewx.auth.dto.ForgotPasswordResponse;
import com.interviewx.auth.dto.GoogleLoginRequest;
import com.interviewx.auth.dto.LoginRequest;
import com.interviewx.auth.dto.LoginResponse;
import com.interviewx.auth.dto.RegisterRequest;
import com.interviewx.auth.dto.RegisterResponse;
import com.interviewx.auth.dto.ResetPasswordRequest;
import com.interviewx.auth.dto.ResetPasswordResponse;
import com.interviewx.auth.dto.VerifyOtpRequest;
import com.interviewx.auth.dto.VerifyOtpResponse;

public interface AuthService {

    RegisterResponse register(RegisterRequest request);

    LoginResponse login(LoginRequest request);

    ForgotPasswordResponse forgotPassword(ForgotPasswordRequest request);
    
    VerifyOtpResponse verifyOtp(VerifyOtpRequest request);

    ResetPasswordResponse resetPassword(
        ResetPasswordRequest request);

        LoginResponse googleLogin(GoogleLoginRequest request);
}